### ISSUE TITLE
Dropdown column header filter contains bad characters upon activation

### ISSUE THREAD

**Issue by thujone** [olifolkerd/tabulator#3107]
**Describe the bug**
I have a column header filter for my column named 'Status'. It's a dropdown with several options: COMPLETED, PROCESSING, FAILED, ERROR, and '' (blank). When you click on the filter, the text in the box is temporarily changed from, for example, 'COMPLETED' to ', COMPLETED'. Or 'PROCESSING' to ', PROCESSING'. Or 'ERROR' to ', ERROR'. 

If you notice the pattern, the dropdown when it's active adds a comma and a space to the text in the filter. After you choose an option, the comma and space are removed. The issue is that while it's in this active state, the results below are cleared out entirely. I'm not sure why this is happening or how to overcome it.

**Tabulator Info**
- Which version of Tabulator are you using?
4.9.1

**Working Example**
https://codepen.io/thujone/pen/WNGQQzN?editors=1010

**To Reproduce**
A step by step guide to recreate the issue in your JS Fiddle or Codepen:
1. Click on the header filter for Status and choose any of the options that aren't blank.
2. Click again on the Status filter to activate the dropdown.
3. You can see that while it's in the active state, all of the results have already disappeared (filtered out) before you even choose another option.

**Expected behavior**
I expected that the currently-active option does not contain a comma and space, and that the results below do not get filtered out until you actually choose another option.

**Screenshots**
None

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Win10 Pro Chrome and Firefox
 - Chrone and Firefox
 - latest versions

**Smartphone (please complete the following information):**
n/a

**Additional context**

**Comment by thujone** [comment]
An update: I have been able to partially fix the issue by removing the empty name-value pair in the list of options. Specifically I changed this column configuration from this...

` { title: 'Status', field: 'status', formatter: 'html', hozAlign: 'center', editorParams: { values: {'':'','COMPLETED':'COMPLETED','PROCESSING':'PROCESSING','ERROR':'ERROR','FAILED':'FAILED'} }, headerFilter: 'select', headerFilterPlaceholder: 'Select', headerFilterParams: { values: {'':'','COMPLETED':'COMPLETED','PROCESSING':'PROCESSING','ERROR':'ERROR','FAILED':'FAILED'} }, multiselect: true }`

...to this:

` { title: 'Status', field: 'status', formatter: 'html', hozAlign: 'center', editorParams: { values: {'COMPLETED':'COMPLETED','PROCESSING':'PROCESSING','ERROR':'ERROR','FAILED':'FAILED'} }, headerFilter: 'select', headerFilterPlaceholder: 'Select', headerFilterParams: { values: {'COMPLETED':'COMPLETED','PROCESSING':'PROCESSING','ERROR':'ERROR','FAILED':'FAILED'} }, multiselect: true }`

Basically I just removed the empty string initial state, and the comma-space problem disappeared. Unfortunately this is an imperfect fix, because I need to allow the user the option to clear out the filter by going back to the empty state if they wish. Not sure how to do this, or if it's even possible.
